File: Forms/formEditor.vb
''' <summary>
''' Description of form/class/etc.
''' </summary>
Public Partial Class formEditor
	Private stringExport(,) As String
	Private iniFilehandler As New iniHandler
	
	Public Sub New()
		' The Me.InitializeComponent call is required for Windows Forms designer support.
		Me.InitializeComponent()
		textboxFilename.Text = System.IO.Directory.GetCurrentDirectory & "\data\"
	End Sub
	
	Sub ButtonCloseClick(sender As Object, e As EventArgs)
		Me.Hide()
	End Sub
	
	Sub ButtonSaveClick(sender As Object, e As EventArgs)
		textboxEdit.AppendText(Chr(13))
		Dim stringMediator() As String = textboxEdit.Text.Split(Chr(10))
		Dim integerCount As Integer = 0
		Do
			stringMediator(integerCount) = stringMediator(integerCount).Remove(stringMediator(integerCount).Length - 1, 1)
			integerCount += 1
		Loop Until integerCount > stringMediator.GetUpperBound(0)
		
		Dim integerColumn As Integer = -1
		Dim integerRow As Integer = 0
		
		Dim stringColumns As String()
		stringColumns = textboxEdit.Text.Split(Chr(91))
		For Each line As String In stringColumns
			Dim stringRow() As String = line.Split(Chr(60))
			If integerRow < stringRow.GetUpperBound(0) Then integerRow = stringRow.GetUpperBound(0)
		Next
		ReDim stringExport(stringColumns.GetUpperBound(0) - 1, (integerRow * 2))
		
		integerRow = 0
		For Each line As String In stringMediator
			If line.StartsWith("[") = True Then
				integerColumn += 1
				integerRow = 0
				stringExport(integerColumn, integerRow) = line
			ElseIf line = "" Then
				
			Else
				integerRow += 1
				stringExport(integerColumn, integerRow) = line
			End If
		Next
		iniFilehandler.Save(textboxFilename.Text, stringExport)
	End Sub
	
	Sub ButtonLoadClick(sender As Object, e As EventArgs)
		textboxEdit.Text = ""
		stringExport = iniFilehandler.Load(textboxFilename.Text)
		For Each line As String In stringExport
			textboxEdit.AppendText(line & Chr(13) & Chr(10))
		Next
	End Sub
	
	Sub ButtonBrowseClick(sender As Object, e As EventArgs)
		Dim filebrowser As New System.Windows.Forms.SaveFileDialog
		filebrowser.AddExtension = False
		filebrowser.DefaultExt = "pd"
		filebrowser.InitialDirectory = System.IO.Directory.GetCurrentDirectory & "\data"
		filebrowser.Title = "Choose new name or edit existing one."
		filebrowser.ShowDialog()
		textboxFilename.Text = filebrowser.FileName
	End Sub
	
	Sub FormEditorLoad(sender As Object, e As EventArgs)
		listboxDescription.Items.Add("Items")
		listboxDescription.Items.Add("Customers")
		listboxDescription.Items.Add("Settings")
	End Sub
	
	Sub ListboxDescriptionSelectedIndexChanged(sender As Object, e As EventArgs)
		If listboxDescription.SelectedIndex = 0 Then 'Items
			textboxDescription.Text = "[ItemInformation]" & Chr(13) & Chr(10) & "<Name/Title>" & Chr(13) & Chr(10) & "<Brand/Author>" & Chr(13) & Chr(10) & "<Company/Publisher>" & Chr(13) & Chr(10) & "<Genre>" & Chr(13) & Chr(10) & "<Category>" & Chr(13) & Chr(10) & "<Family>" & Chr(13) & Chr(10) & Chr(13) & Chr(10) & "[Statistics]" & Chr(13) & Chr(10) & "<Quality>" & Chr(13) & Chr(10) & "<Popularity>" & Chr(13) & Chr(10) & "<Rarity>" & Chr(13) & Chr(10) & "<BaseWorth>" & Chr(13) & Chr(10) & "<ItemTier>" & Chr(13) & Chr(10) & Chr(13) & Chr(10) & "[Various]" & Chr(13) & Chr(10) & "<PictureFilename>" & Chr(13) & Chr(10) & "<Description>"
		ElseIf listboxDescription.SelectedIndex = 1 Then 'Customers
			textboxDescription.Text = "[Age]" & Chr(13) & Chr(10) & "<Min>" & Chr(13) & Chr(10) & "<Max>" & Chr(13) & Chr(10) & Chr(13) & Chr(10) & "[Money]" & Chr(13) & Chr(10) & "<Min>" & Chr(13) & Chr(10) & "<Max>" & Chr(13) & Chr(10) & Chr(13) & Chr(10) & "[Desire]" & Chr(13) & Chr(10) & "<Min>" & Chr(13) & Chr(10) & "<Max>"
		ElseIf listboxDescription.SelectedIndex = 2 Then 'Settings
			textboxDescription.Text = "[WarningMessages]" & Chr(13) & Chr(10) & "<ProgramQuit>" & Chr(13) & Chr(10) & "<NewgameOverwrite>" & Chr(13) & Chr(10) & "<OptionsApplyRestart>"
		End If
	End Sub
End Class

File: Game/Forms/formEditor.cs
using Microsoft.VisualBasic;
using System;

/// <summary>
/// Description of form/class/etc.
/// </summary>
public partial class formEditor
{
    private string[,] stringExport;
    private iniHandler iniFilehandler = new iniHandler();

    public formEditor()
    {
        // The Me.InitializeComponent call is required for Windows Forms designer support.
        this.InitializeComponent();
        textboxFilename.Text = System.IO.Directory.GetCurrentDirectory() + "\\data\\";
    }

    public void buttonClose_Click(object sender, EventArgs e)
    {
        this.Hide();
    }

    public void buttonSave_Click(object sender, EventArgs e)
    {
        textboxEdit.AppendText(((char)13).ToString());
        string[] stringMediator = textboxEdit.Text.Split((char)10);
        int integerCount = 0;
        do {
            stringMediator[integerCount] = stringMediator[integerCount].Remove(stringMediator[integerCount].Length - 1, 1);
            integerCount += 1;
        } while (!(integerCount > stringMediator.GetUpperBound(0)));

        int integerColumn = -1;
        int integerRow = 0;

        string[] stringColumns = null;
        stringColumns = textboxEdit.Text.Split((char)91);
        foreach (string line in stringColumns) {
            string[] stringRow = line.Split((char)60);
            if (integerRow < stringRow.GetUpperBound(0))
                integerRow = stringRow.GetUpperBound(0);
        }
        stringExport = new string[stringColumns.GetUpperBound(0), (integerRow * 2) + 1];

        integerRow = 0;
        foreach (string line in stringMediator) {
            if (line.StartsWith(((char)91).ToString())) {
                integerColumn += 1;
                integerRow = 0;
                stringExport[integerColumn, integerRow] = line;
            } else if (string.IsNullOrEmpty(line)) {

            } else {
                integerRow += 1;
                stringExport[integerColumn, integerRow] = line;
            }
        }
        iniFilehandler.Save(textboxFilename.Text, stringExport);
    }

    public void buttonLoad_Click(object sender, EventArgs e)
    {
        textboxEdit.Text = "";
        stringExport = iniFilehandler.Load(textboxFilename.Text);
        foreach (string line in stringExport) {
            textboxEdit.AppendText(line + (char)13 + (char)10);
        }
    }

    public void buttonBrowse_Click(object sender, EventArgs e)
    {
        System.Windows.Forms.SaveFileDialog filebrowser = new System.Windows.Forms.SaveFileDialog();
        filebrowser.AddExtension = false;
        filebrowser.DefaultExt = "pd";
        filebrowser.InitialDirectory = System.IO.Directory.GetCurrentDirectory() + "\\data";
        filebrowser.Title = "Choose new name or edit existing one.";
        filebrowser.ShowDialog();
        textboxFilename.Text = filebrowser.FileName;
    }

    public void formEditor_Load(object sender, EventArgs e)
    {
        listboxDescription.Items.Add("Items");
        listboxDescription.Items.Add("Customers");
        listboxDescription.Items.Add("Settings");
        listboxDescription.Items.Add("Namelist");
    }

    public void listboxDescription_SelectedIndexChanged(object sender, EventArgs e)
    {
        switch (listboxDescription.SelectedIndex) {
            case 0:
                //Items
                textboxDescription.Text = "[ItemInformation]" + (char)13 + (char)10 + "<Name/Title>" + (char)13 + (char)10 + "<Brand/Author>" + (char)13 + (char)10 + "<Company/Publisher>" + (char)13 + (char)10 + "<Genre>" + (char)13 + (char)10 + "<Category>" + (char)13 + (char)10 + "<Family>" + (char)13 + (char)10 + (char)13 + (char)10 + "[Statistics]" + (char)13 + (char)10 + "<Quality>" + (char)13 + (char)10 + "<Popularity>" + (char)13 + (char)10 + "<Rarity>" + (char)13 + (char)10 + "<BaseWorth>" + (char)13 + (char)10 + "<ItemTier>" + (char)13 + (char)10 + (char)13 + (char)10 + "[Various]" + (char)13 + (char)10 + "<PictureFilename>" + (char)13 + (char)10 + "<Description>";
                break;
            case 1:
                //Customers
                textboxDescription.Text = "[Age]" + (char)13 + (char)10 + "<Min>" + (char)13 + (char)10 + "<Max>" + (char)13 + (char)10 + (char)13 + (char)10 + "[Money]" + (char)13 + (char)10 + "<Min>" + (char)13 + (char)10 + "<Max>" + (char)13 + (char)10 + (char)13 + (char)10 + "[Desire]" + (char)13 + (char)10 + "<Min>" + (char)13 + (char)10 + "<Max>";
                break;
            case 2:
                //Settings
                textboxDescription.Text = "[WarningMessages]" + (char)13 + (char)10 + "<LastProfileID>" + (char)13 + (char)10 + "<ProgramQuit>" + (char)13 + (char)10 + "<NewgameOverwrite>" + (char)13 + (char)10 + "<OptionsApplyRestart>";
                break;
            case 3:
                //Namelist
                textboxDescription.Text = "[Name List Data]" + (char)13 + (char)10 + "<Male>" + (char)13 + (char)10 + "<Female>";
                break;
        }
    }
}